File: python/flower_bed.py
# https://leetcode.com/problems/can-place-flowers/

class Solution:
    # if previous not found but next is 0 == placed
    # if next is not found but previous is 0 == placed
    # if previous and next is 0 == placed
    def canPlaceFlowers(self, flowerbed, n):
        placed = 0
        end_position = len(flowerbed)
        last_placed = False
        for current in range(end_position):
            if placed == n:
                return True
            
            value = flowerbed[current]
            if value == 1 or last_placed:
                last_placed = False
                continue

            previous = value if current < 1 else flowerbed[current - 1]
            nxxt = value if current == (end_position - 1) else flowerbed[current + 1]

            if (previous == 0 and nxxt == 0):
                placed += 1
                last_placed = True

        return placed == n

File: javascript/flower_bed.js
// https://leetcode.com/problems/can-place-flowers/

/**
 * @param {number[]} flowerbed
 * @param {number} n
 * @return {boolean}
 */
var canPlaceFlowers = function(flowerbed, n) {
    let placed = 0;
    const endPosition = flowerbed.length;
    let lastPlaced = false;

    for (current = 0; current < endPosition; current++) { 
        if (placed == n) {
            return true;
        }

        value = flowerbed[current];
        if (value || lastPlaced) {
            lastPlaced = false;
            continue;
        }

        previous = !current ? value : flowerbed[current - 1];
        nxxt = current == (endPosition - 1) ? value : flowerbed[current + 1];

        if (!previous && !nxxt) {
            placed += 1;
            lastPlaced = true;
        }
    }

    return placed == n;
}

flowerbed = [1,0,0,0,1]
n = 1
x = canPlaceFlowers(flowerbed, n);
console.log(x);